### PR TITLE
Issue#431 upload image in featured content

### PR DESCRIPTION
**Pull request by Mkohm**
Resolves https://github.com/GlobalDigitalLibraryio/issues/issues/431

### CODE REVIEW

**Comment by Mkohm** [comment]
@alexanbj klar for review!

File: packages/admin-frontend/components/FileDialog.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    return (
+      <Dialog open={this.props.fileDialogOpen} onClose={this.props.onCancel}>
+        <DialogTitle>Upload file</DialogTitle>
+        <DialogContent>
+          <p>
+            We can create a form here asking for values like alttext or other
+            metadata
+          </p>
+          <img
+            src={URL.createObjectURL(this.props.selectedFile)}
```
Comment (alexanbj) [inline packages/admin-frontend/components/FileDialog.js]: See memory management notes here: https://developer.mozilla.org/en-US/docs/Web/API/URL/createObjectURL

File: packages/admin-frontend/pages/featured.js [hunk 2]
```diff
@@ -251,26 +287,73 @@ export default class EditFeaturedContent extends React.Component<Props, State> {
...
+                      )}
+                    />
+                  </div>
+
+                  <div css={{ margin: '20px' }}>or</div>
+
+                  <div css={{ align: 'center' }}>
+                    <Field
+                      name="selectedFile"
+                      render={({ input }) => (
```
Comment (alexanbj) [inline packages/admin-frontend/pages/featured.js]: I don't think the file should be part of the form state? Looks like it isn't tied to the form (but a separate upload step). But the file is still in the page state and the form state?

File: packages/admin-frontend/pages/featured.js [hunk 3]
```diff
@@ -146,6 +153,35 @@ export default class EditFeaturedContent extends React.Component<Props, State> {
...
+  handleOnCancel = (change: (value: any) => void) => {
+    this.setState({ fileDialogOpen: false });
+
+    // Set the value of the file-input to the empty string to reset it
+    change('');
+  };
+
+  handleFileChosen = (
+    event: SyntheticInputEvent<EventTarget>,
+    onChange: (SyntheticInputEvent<any> | any) => void
```
Comment (alexanbj) [inline packages/admin-frontend/pages/featured.js]: syntethicinputevent or any?

File: packages/admin-frontend/pages/featured.js (hunk 2 above)
Comment (Mkohm) [reply packages/admin-frontend/pages/featured.js]: Det var slik jeg tenkte før jeg startet å implementere dette, men for at man skal kunne trykke "discard changes" og resette form ved endringer så må jeg inkludere valgt fil i formet. På onSubmit så blir da denne verdien slettet slikt at det ikke blir med i POST/PUT.

File: packages/admin-frontend/pages/featured.js (hunk 3 above)
Comment (Mkohm) [reply packages/admin-frontend/pages/featured.js]: Ser at dette burde være `SyntheticInputEvent<*> | any => void` ifølge: https://github.com/final-form/react-final-form#inputonchange-syntheticinputevent--any--void

**Pull request merged**